experimental/dds/tree2/src/feature-libraries/flex-tree/lazyNode.ts

```diff
@@ -331,7 +331,7 @@ export abstract class LazyTreeNode<TSchema extends TreeNodeSchema = TreeNodeSche
 		if (event.propagationStopped) {
 			return;
 		}
-		const parentNode = this.parentField.parent.parent;
+		const parentNode = this.#anchorNode.parent?.slots.get(lazyTreeSlot);
```

Comment:
This change means that if no flex tree node has been created for the parent, it will be undefined, instead of just the root case. This is possible to happen in some edge cases, like the child node was moved (by a remote client) under a node the local client has never accessed through the flex tree API.

This might be a valid optimization, or it might be a bug: I can tell, but it should probably be changed (if not intended) or commented (if intended) since its pretty subtle.

Reply:
If you are having issues in this code, adding some asserts that the node isn't deleted might detect some bad states: real strange things can happen if you try and use deleted nodes and the symptoms can be real confusing when they don't assrt.